imitation/scripts/train.py

```diff
@@ -104,21 +108,22 @@ def disc_plot_show():
   def ep_reward_plot_add_data(env, name):
     """Calculate and record the average episode reward from rollouts of env."""
     gen_policy = trainer.gen_policy
-    rand_policy = util.make_blank_policy(trainer.env)
-    exp_policy = trainer.expert_policies[-1]
-
     gen_ret = util.rollout.mean_return(
         gen_policy, env, n_episodes=n_episodes_per_reward_data)
+    gen_ep_reward[name].append(gen_ret)
+    tf.logging.info("generator return: {}".format(gen_ret))
+
+    rand_policy = util.make_blank_policy(trainer.env)
     rand_ret = util.rollout.mean_return(
         rand_policy, env, n_episodes=n_episodes_per_reward_data)
-    exp_ret = util.rollout.mean_return(
-        exp_policy, env, n_episodes=n_episodes_per_reward_data)
-    gen_ep_reward[name].append(gen_ret)
     rand_ep_reward[name].append(rand_ret)
-    exp_ep_reward[name].append(exp_ret)
-    tf.logging.info("generator return: {}".format(gen_ret))
     tf.logging.info("random return: {}".format(rand_ret))
-    tf.logging.info("exp return: {}".format(exp_ret))
+
+    if expert_policy is not None:
```

Comment:
It would also be meaningful to report the reward of the demonstration rollouts. In the true reward case, we can just add rewards to the saved trajectories. In the learned reward, we can directly query the reward model (no need for rollouts). It'd be interested in the learned reward case to see if they diverge, since the demonstrations are training data and here we're computing new expert rollouts.

Reply:
Good idea, I'll add an issue for this later today